src/python/py/models/builder.py

```diff
@@ -292,6 +292,7 @@ def __init__(self, config, io_dtype, onnx_dtype, ep, cache_dir, extra_options):
                 "is_symmetric": extra_options.get("int4_is_symmetric", True),
                 "op_types_to_quantize": extra_options.get("int4_op_types_to_quantize", ("MatMul", )),
                 "nodes_to_exclude": extra_options.get("int4_nodes_to_exclude", []),
+                "quant_method": extra_options.get("quant_method", "rtn"),
```

Comment:
The keys in `self.quant_attrs["int4"]` should match the same names that the constructor for `MatMulNBitsQuantizer` accepts. The keys in `extra_options` are prefixed with `int4` to clarify that they are specifically used for int4 quantization.

```suggestion
                "algo_config": self.make_int4_algo_config(extra_options.get("int4_algo_config", "default")) ,
```

If/when quantization is supported for `n` bits, the `int4` prefixes in the `extra_options` can be removed accordingly.

Reply:
`algo_config` is the actual RTNWeightOnlyQuantConfig that generated from `quant_method`. The users are providing `quant_method` and we converted it into the actual Config.